### ISSUE TITLE
[mobx-undecorate] fail with different errors

### ISSUE THREAD

**Issue by AoDev** [mobxjs/mobx#2501]
**Intended outcome:**

Migrating from Mobx 4 to Mobx 6 using mobx-undecorate. 
* Pure ES6 codebase. No typescript. 
* Pure External stores (no state in UI components). 

Simple classes like: 

```js
import {observable, action} from 'mobx'

/**
 * Model for the user
 */
export default class UserModel {
  @observable id = ''
  @observable name = ''

  @action.bound set (prop, value) {
    this[prop] = value
  }

  @action.bound assign (props) {
    Object.assign(this, props)
  }

  @action.bound async updateData (updatedUserData) {
    const userData = await myApi.getUser()
    // Note: possibility to use observable patterns for automatic updates
    this.assign(userData)
  }

  constructor (userData) {
    this.assign(userData)
  }
}

```
**Actual outcome:**

* Tons of the same errors listed below during the process. Nothing is transformed.
* Thing I have noticed is that it seems to scan `node_modules`. I think it should ignore it.

## Error 1: "non-mobx decorator"
```
[mobx:undecorate] Found non-mobx decorator @observable at ([edited]/UserModel.js:9:2):
        @observable name = ''
        ^

[mobx:undecorate] Found non-mobx decorator @computed at ([edited]/shared-components/viewModels/List.js:15:2):
        @computed get filteredItems () {
        ^

```

## Error 2
```
Transformation error (Unexpected token, expected "," (29:14))
SyntaxError: Unexpected token, expected "," (29:14)
    at Object._raise ([edited]/11981/lib/node_modules/mobx-undecorate/node_modules/@babel/parser/src/parser/error.js:60:45)
    at Object.raiseWithData ([edited]/11981/lib/node_modules/mobx-undecorate/node_modules/@babel/parser/src/parser/error.js:55:17)
    at Object.raise ([edited]/11981/lib/node_modules/mobx-undecorate/node_modules/@babel/parser/src/parser/error.js:39:17)
    at Object.unexpected ([edited]/11981/lib/node_modules/mobx-undecorate/node_modules/@babel/parser/src/parser/util.js:139:16)
    at Object.expect ([edited]/11981/lib/node_modules/mobx-undecorate/node_modules/@babel/parser/src/parser/util.js:116:28)
    at Object.tsParseDelimitedListWorker ([edited]/11981/lib/node_modules/mobx-undecorate/node_modules/@babel/parser/src/plugins/typescript/index.js:272:16)
    at Object.tsParseDelimitedList ([edited]/11981/lib/node_modules/mobx-undecorate/node_modules/@babel/parser/src/plugins/typescript/index.js:232:14)
    at Object.tsParseBracketedList ([edited]/11981/lib/node_modules/mobx-undecorate/node_modules/@babel/parser/src/plugins/typescript/index.js:294:27)
    at Object.tsParseTypeParameters ([edited]/11981/lib/node_modules/mobx-undecorate/node_modules/@babel/parser/src/plugins/typescript/index.js:394:26)
    at [edited]/11981/lib/node_modules/mobx-undecorate/node_modules/@babel/parser/src/plugins/typescript/index.js:2405:31
```

**How to reproduce the issue:**

Just followed the instructions, modified a couple of things in babel config, ran the tool in the root folder.

**Versions**
node v14.13.0
latest mobx-undecorate

**Comment by mweststrate** [comment]
For the first 2 problems, see the docs of
https://www.npmjs.com/package/mobx-undecorate

For the node_modules scan; run the command in your source folder, not in
the project root .

For the non-mobx decorator found, you might want to try the --ignoreImports
flag

For Error 2 we need a specific source file (please try to trim it down to
the smallest file that still yields the error) and your babel config in
order to be able to look into it

**Comment by AoDev** [comment]
Thanks for the tips 👍 

## First issue
I don't understand what the `--ignoreImports` really mean. 
> the codemod will only convert decorators if they are imported from a MobX package

All the stores / models in the codebase either import everything as mobx or import a named export: 
* `import * as mobx from 'mobx'`
* `import {observable} from 'mobx'`

Isn't this the definition of "imported from a mobx package" ?

### makeObservable is not imported
* expected: `makeObservable` should have been imported through the codemod.

<img width="468" alt="Screenshot 2020-10-10 at 9 07 48" src="https://user-images.githubusercontent.com/1526150/95648513-18c96e80-0ad8-11eb-81ca-666e27e6676a.png">

## About the second issue
`Transformation error (Unexpected token, expected "," (29:14))` (...)

I found that it happens with React components.

A simple component like this will trigger the error:

```jsx
import React from 'react'
import {observer} from 'mobx-react'

export function MyComponent () {
  return <div/>
}

export default observer(MyComponent)

```

**Comment by mweststrate** [comment]
Hmm I'm suddenly wondering if JSX parsing is enabled in the babel config of the codemod. Probably it should just be added here https://github.com/mobxjs/mobx/blob/mobx6/packages/mobx-undecorate/src/undecorate.ts#L24. I'll only have time to look into that on Thursday or something, so if anyone beats me to it feel free to add it. Please do include a unit tests. 

The imports not working is more surprising, as that part is covert by unit tests.

**Comment by mweststrate** [comment]
Released version 1.0.2. to fix the JSX parsing issue, and expanded the readme to clarify the `--ignoreImports`
